src/Esprima/Ast/List.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Esprima.Ast
+{
+    // This structure is like List<> from the BCL except it is designed
```

Comment:
nit: should this be proper `/// <summary>` kind of comment?

Reply:
It's note for library authors and immaterial for the public. A public summary would just say it's a list/collection of `T`.